builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -421,9 +421,16 @@ func resourceVSphereVirtualMachine() *schema.Resource {
 							Default:  "scsi",
 							ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
 								value := v.(string)
-								if value != "scsi" && value != "ide" {
```

Comment:
Hey @dkalleg 

Just a FYI, I wrote a validateFunc that looks like this:

```
func validateArmStorageBlobType(v interface{}, k string) (ws []string, errors []error) {
	value := strings.ToLower(v.(string))
	validTypes := map[string]struct{}{
		"blob": struct{}{},
		"page": struct{}{},
	}

	if _, ok := validTypes[value]; !ok {
		errors = append(errors, fmt.Errorf("Blob type %q is invalid, must be %q or %q", value, "blob", "page"))
	}
	return
}
```

Notice the use of map and not having to iterate over it ;)

P.